kolibri/plugins/safe_html5_viewer/assets/src/views/SafeHtml5RendererIndex.vue

```diff
@@ -120,12 +131,13 @@
         if (this.forceDurationBasedProgress) {
           progress = this.durationBasedProgress;
         } else {
-          // TODO: Handle progress tracking based on how
-          // much of the article has been scrolled through
+          // Use scroll events to track progress
           progress = this.scrollBasedProgress;
         }
         this.$emit('updateProgress', progress);
-        if (progress >= 1) {
```

Comment:
I wonder if we could maintain this strictness here - but instead add a correction during the scroll based calculation instead? Otherwise we may end up with a lot of content marked as in progress when a learner thinks they have finished (the `finished` event doesn't trigger anything progress tracking wise, it's used to disambiguate the technical completion from when it is desirable to trigger further follow up for a user - like displaying the completion modal).

Reply:
Makes sense